diff --git a/node_modules/expo/scripts/autolinking.gradle b/node_modules/expo/scripts/autolinking.gradle
index 929b7f0..3b1a09c 100644
--- a/node_modules/expo/scripts/autolinking.gradle
+++ b/node_modules/expo/scripts/autolinking.gradle
@@ -1,5 +1,6 @@
 // Resolve `expo` > `expo-modules-autolinking` dependency chain
-def autolinkingPath = ["node", "--print", "require.resolve('expo-modules-autolinking/package.json', { paths: [require.resolve('expo/package.json')] })"]
+def nodeBinary = providers.gradleProperty("expo.nodeExecutable").getOrElse("/usr/local/bin/node")
+def autolinkingPath = [nodeBinary, "--print", "require.resolve('expo-modules-autolinking/package.json', { paths: [require.resolve('expo/package.json')] })"]
 apply from: new File(
   providers.exec {
     workingDir(rootDir)
